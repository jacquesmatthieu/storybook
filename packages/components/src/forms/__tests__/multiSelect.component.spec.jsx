import React from 'react';
import { shallow } from 'enzyme';

import MultiSelect from '../multiSelect.component';

describe('Dropdown', () => {
  let props;

  beforeEach(() => {
    props = {
      input: {
        value: ['foo'],
        name: 'foo',
        onChange: jest.fn(),
      },
      meta: {},
      options: [
        ['foo', 'Foo'],
        ['bar', 'Bar'],
      ],
    };
  });

  const getWrapper = () => shallow(<MultiSelect {...props} />);

  it('should render with default props', () => {
    // When
    const wrapper = getWrapper();
    const subWrapper = wrapper.find('multiSelectcomponent__WrapperDropdown')
      .renderProp('children', { open: false, toggle: jest.fn() });

    // Then
    expect(wrapper).toMatchSnapshot();
    expect(subWrapper).toMatchSnapshot();
  });

  it('should render with specified props', () => {
    // Given
    props.label = 'bar';
    props.meta = {
      error: true,
      touched: true,
      valid: true,
    };

    // When
    const wrapper = getWrapper();
    const subWrapper = wrapper.find('multiSelectcomponent__WrapperDropdown')
      .renderProp('children', { open: true, toggle: jest.fn() });

    // Then
    expect(wrapper).toMatchSnapshot();
    expect(subWrapper).toMatchSnapshot();
  });
});
