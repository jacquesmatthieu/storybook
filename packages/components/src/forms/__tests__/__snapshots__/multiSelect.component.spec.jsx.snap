// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render with default props 1`] = `
<InputGroup
  className=""
  fullWidth={false}
  label=""
  name="foo"
>
  <multiSelectcomponent__WrapperDropdown>
    <Component />
  </multiSelectcomponent__WrapperDropdown>
</InputGroup>
`;

exports[`Dropdown should render with default props 2`] = `
<Fragment>
  <multiSelectcomponent__InvisibleInput />
  <multiSelectcomponent__StyledMultiSelect
    disabled={false}
    error={false}
    innerRef={[Function]}
    onClick={[MockFunction]}
    onKeyDown={[Function]}
    open={false}
    tabIndex={0}
  >
    Foo
    <multiSelectcomponent__SelectIcon
      direction="down"
      name="arrow"
      size={15}
      stroke="grey.dark"
      strokeWidth={1}
    />
  </multiSelectcomponent__StyledMultiSelect>
  <multiSelectcomponent__StyledMultiSelectDropdown
    open={false}
  >
    <Checkbox
      checked={true}
      key="foo"
      name="-foo"
      onChange={[Function]}
      selected={true}
      type="checkbox"
      value="foo"
    >
      Foo
    </Checkbox>
    <Checkbox
      checked={false}
      key="bar"
      name="-bar"
      onChange={[Function]}
      selected={false}
      type="checkbox"
      value="bar"
    >
      Bar
    </Checkbox>
  </multiSelectcomponent__StyledMultiSelectDropdown>
</Fragment>
`;

exports[`Dropdown should render with specified props 1`] = `
<InputGroup
  className=""
  fullWidth={false}
  label="bar"
  name="foo"
>
  <multiSelectcomponent__WrapperDropdown>
    <Component />
  </multiSelectcomponent__WrapperDropdown>
</InputGroup>
`;

exports[`Dropdown should render with specified props 2`] = `
<Fragment>
  <multiSelectcomponent__InvisibleInput />
  <multiSelectcomponent__StyledMultiSelect
    disabled={false}
    error={true}
    innerRef={[Function]}
    onClick={[MockFunction]}
    onKeyDown={[Function]}
    open={true}
    tabIndex={0}
    valid={true}
  >
    Foo
    <multiSelectcomponent__SelectIcon
      direction="up"
      name="arrow"
      size={15}
      stroke="grey.dark"
      strokeWidth={1}
    />
  </multiSelectcomponent__StyledMultiSelect>
  <multiSelectcomponent__StyledMultiSelectDropdown
    open={true}
  >
    <Checkbox
      checked={true}
      key="foo"
      name="bar-foo"
      onChange={[Function]}
      selected={true}
      type="checkbox"
      value="foo"
    >
      Foo
    </Checkbox>
    <Checkbox
      checked={false}
      key="bar"
      name="bar-bar"
      onChange={[Function]}
      selected={false}
      type="checkbox"
      value="bar"
    >
      Bar
    </Checkbox>
  </multiSelectcomponent__StyledMultiSelectDropdown>
</Fragment>
`;
